bottlecap/src/traces/trace_agent.rs

```diff
@@ -405,7 +411,15 @@ impl TraceAgent {
         invocation_processor: Arc<Mutex<InvocationProcessor>>,
         tags_provider: Arc<provider::Provider>,
         version: ApiVersion,
+        api_key_factory: Arc<ApiKeyFactory>,
     ) -> Response {
+        let Some(api_key) = api_key_factory.get_api_key().await else {
```

Comment:
Abort at the beginning of v0.4 and v0.5 traces API handler

Reply:
dit: Is it possible that the customer's code calls traces api synchronously, and we slow down the customer's Lambda by doing the heavy operation of resolving api key here?
If so, it might be better to further defer key resolution by moving it out of the API handler.

Reply:
yeah we need to defer this to the flusher as this API is called synchronously